source/utilities.tex

```diff
@@ -13429,6 +13443,355 @@
 \returns
 \tcode{true} if \tcode{f} has no target object, otherwise \tcode{false}.
 \end{itemdescr}
+
+\rSec3[func.wrap.ref]{Non-owning wrapper}
+
+\rSec4[func.wrap.ref.general]{General}
+
+\pnum
+The header provides partial specializations of \tcode{function_ref}
+for each combination of the possible replacements of
+the placeholders \cv{} and \placeholder{noex} where:
+
+\begin{itemize}
+\item \cv{} is either const or empty, and
+\item \placeholder{noex} is either \tcode{true} or \tcode{false}.
+\end{itemize}
+
+\rSec4[func.wrap.ref.class]{Class template \tcode{function_ref}}
+
+\indexlibraryglobal{function_ref}%
+\begin{codeblock}
+namespace std {
+  template<class R, class... ArgTypes>
+  class function_ref<R(ArgTypes...) @\cv{}@ noexcept(@\placeholder{noex}@)> {
+  public:
+    // \ref{func.wrap.ref.ctor}, constructors and assignment operators
+    template<class F> function_ref(F*) noexcept;
+    template<class F> constexpr function_ref(F&&) noexcept;
+    template<auto f> constexpr function_ref(nontype_t<f>) noexcept;
+    template<auto f, class U>
+      constexpr function_ref(nontype_t<f>, U&&) noexcept;
+    template<auto f, class T>
+      constexpr function_ref(nontype_t<f>, @\cv{}@ T*) noexcept;
+
+    constexpr function_ref(const function_ref&) noexcept = default;
+    constexpr function_ref& operator=(const function_ref&) noexcept = default;
+    template<class T> function_ref& operator=(T) = delete;
+
+    // \ref{func.wrap.ref.inv}, invocation
+    R operator()(ArgTypes...) const noexcept(@\placeholder{noex}@);
+
+  private:
+    template<class... T>
+      static constexpr bool @\exposid{is-invocable-using}@ = @\seebelow@;     // \expos
+  };
+
+  // \ref{func.wrap.ref.deduct}, deduction guides
+  template<class F>
+    function_ref(F*) -> function_ref<F>;
+  template<auto f>
+    function_ref(nontype_t<f>) -> function_ref<@\seebelow@>;
+  template<auto f>
+    function_ref(nontype_t<f>, auto) -> function_ref<@\seebelow@>;
```

Comment:
See <https://github.com/cplusplus/draft/issues/6362#issuecomment-1612478224>.
```suggestion
  template<auto f, class T>
    function_ref(nontype_t<f>, T&&) -> function_ref<@\seebelow@>;
```

Reply:
@burblebee , we need that fix, preferably in a separate commit.

Reply:
After the motions, thanks!